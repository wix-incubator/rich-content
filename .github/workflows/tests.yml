name: tests
on: push
env:
  captureVideos: 'true'
  captureImages: 'true'
  APPLITOOLS_BATCH_ID: ${{ github.sha }}

jobs:
  cleanup-runs:
    runs-on: ubuntu-latest
    steps:
      - uses: rokroskar/workflow-run-cleanup-action@v0.2.2
        if: "!startsWith(github.ref, 'refs/tags/') && github.ref != 'refs/heads/master'"
        env:
          GITHUB_TOKEN: '${{ secrets.GITHUB_TOKEN }}'

  e2e-visual:
    runs-on: ubuntu-18.04
    strategy:
      fail-fast: false
      matrix:
        ci_node_total: [5] # Set N number of parallel jobs you want to run tests on.
        ci_node_index: [0, 1, 2, 3, 4] # set N-1 indexes for parallel jobs

    steps:
      - uses: actions/checkout@v2
      - id: cache
        uses: actions/cache@master
        with:
          path: |
            node_modules
            */*/node_modules
            ~/.cache
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
      - name: npm install + bootstrap
        if: steps.cache.outputs.cache-hit != 'true'
        run: npm ci
      - name: lerna link
        if: steps.cache.outputs.cache-hit == 'true'
        run: npm run link
      - name: build
        run: npm run build:test
      - name: Run http server with the app in the background
        run: |
          npm run e2e:app:run & npm run e2e:app:wait

      - name: Run tests with Knapsack Pro
        env:
          KNAPSACK_PRO_TEST_SUITE_TOKEN_CYPRESS: ${{ secrets.KNAPSACK_PRO_TEST_SUITE_TOKEN_CYPRESS }}
          KNAPSACK_PRO_CI_NODE_BUILD_ID: $(openssl rand -base64 32)
          KNAPSACK_PRO_CI_NODE_TOTAL: ${{ matrix.ci_node_total }}
          KNAPSACK_PRO_CI_NODE_INDEX: ${{ matrix.ci_node_index }}
          KNAPSACK_PRO_TEST_FILE_PATTERN: 'e2e/tests/*.e2e.js'
          APPLITOOLS_API_KEY: ${{ secrets.APPLITOOLS_API_KEY }}
          APPLITOOLS_SERVER_URL: ${{ secrets.APPLITOOLS_SERVER_URL }}
        run: |
          $(npm bin)/knapsack-pro-cypress
      - uses: actions/upload-artifact@v1
        if: failure()
        with:
          name: cypress-screenshots-${{ matrix.ci_node_index }}
          path: e2e/cypress/screenshots
      - uses: actions/upload-artifact@v1
        if: failure()
        with:
          name: cypress-videos-visual-${{ job.status }}-${{ matrix.ci_node_index }}
          path: e2e/cypress/videos

  e2e-visual-applitoolsSendCompletion:
    needs: e2e-visual
    if: always()
    runs-on: ubuntu-18.04
    steps:
      - name: Update Applitools batch status
        uses: wei/curl@v1.1.1
        with:
          args: -d "" -X POST https://wixeyesapi.applitools.com/api/externals/github/servers/github.com/commit/${{ env.APPLITOOLS_BATCH_ID }}/complete?apiKey=${{ secrets.APPLITOOLS_API_KEY }}

  content-state-tests:
    runs-on: ubuntu-18.04
    strategy:
      fail-fast: false
      matrix:
        ci_node_total: [3] # Set N number of parallel jobs you want to run tests on.
        ci_node_index: [0, 1, 2] # set N-1 indexes for parallel jobs

    steps:
      - uses: actions/checkout@v2
      - name: cache
        id: cache
        uses: actions/cache@master
        with:
          path: |
            node_modules
            */*/node_modules
            ~/.cache
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
      - name: npm install + bootstrap
        if: steps.cache.outputs.cache-hit != 'true'
        run: npm ci
      - name: lerna link
        if: steps.cache.outputs.cache-hit == 'true'
        run: npm run link
      - name: build
        run: npm run build:test
      - name: Run http server with the app in the background
        run: npm run e2e:app:run & npm run e2e:app:wait

      - name: Run tests with Knapsack Pro
        env:
          KNAPSACK_PRO_TEST_SUITE_TOKEN_CYPRESS: ${{ secrets.KNAPSACK_PRO_TEST_SUITE_TOKEN_CYPRESS }}
          KNAPSACK_PRO_CI_NODE_BUILD_ID: $(openssl rand -base64 32)
          KNAPSACK_PRO_CI_NODE_TOTAL: ${{ matrix.ci_node_total }}
          KNAPSACK_PRO_CI_NODE_INDEX: ${{ matrix.ci_node_index }}
          KNAPSACK_PRO_TEST_FILE_PATTERN: 'e2e/tests/*.e2e.js'
          APPLITOOLS_IS_DISABLED: true
        run: |
          $(npm bin)/knapsack-pro-cypress --env MATCH_CONTENT_STATE=true
      - uses: actions/upload-artifact@v1
        if: failure()
        with:
          name: cypress-screenshots-${{ matrix.ci_node_index }}
          path: e2e/cypress/screenshots
      - uses: actions/upload-artifact@v1
        if: failure()
        with:
          name: cypress-videos-contentState-${{ job.status }}-${{ matrix.ci_node_index }}
          path: e2e/cypress/videos

  lint-and-unit-tests:
    runs-on: ubuntu-18.04
    steps:
      - uses: actions/checkout@v2
      - name: cache
        id: cache
        uses: actions/cache@master
        with:
          path: |
            node_modules
            */*/node_modules
            ~/.cache
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
      - name: npm install + bootstrap
        if: steps.cache.outputs.cache-hit != 'true'
        run: npm ci
      - name: lerna link
        if: steps.cache.outputs.cache-hit == 'true'
        run: npm run link

      - run: npm run lint
      - run: npm run build
      - run: npm run test:unit

  notify-on-slack:
    runs-on: ubuntu-18.04
    needs: [e2e-visual, content-state-tests, lint-and-unit-tests]
    if: always()
    steps:
      - name: Get commit author
        id: commit-author
        run: |
          echo "::set-env name=commit-author::$(curl -s https://api.github.com/repos/wix-incubator/rich-content/commits/${{ github.sha }} | jq '.author.login' | sed 's/\"//g')"
      - name: Get PR number
        id: pr-number
        run: |
          echo "::set-env name=pr-number::$(curl -s https://api.github.com/repos/wix-incubator/rich-content/commits/${{ github.sha }} | jq '.commit.message' | sed 's/.*(#//g' | sed 's/[^0-9]//g')"
      - name: Get PR author
        id: pr-author
        run: |
          echo "::set-env name=pr-author::$(curl -s https://api.github.com/repos/wix-incubator/rich-content/pulls/${{ env.pr-number }} | jq '.merged_by.login' | sed 's/\"//g')"
      - name: Notify fail in Slack
        uses: voxmedia/github-action-slack-notify-build@v1.1.1
        if: github.ref == 'refs/heads/master' && (needs.e2e-visual.result == 'failure' || needs.content-state-tests.result == 'failure' || needs.lint-and-unit-tests.result == 'failure')
        with:
          channel_id: C0124DH7HHS #rich-content-ci
          status: FAILED ${{ env[env.commit-author] }} ${{ env[env.pr-author] }}
          color: danger

      # - name: Get current commit status
      #   id: current-status
      #   run: |
      #     echo "::set-output name=current-status::$(echo $(curl -s https://api.github.com/repos/wix-incubator/rich-content/commits/${{ github.sha }}/check-suites\?app_id\=15368 -H "Accept: application/vnd.github.antiope-preview+json" | jq '.check_suites | [.[] | select( .head_branch == "master") | .conclusion] | .[:2] | .[]' | sed 's/\"//g'))"
      - name: Get previous commit
        id: previous-commit
        run: |
          echo "::set-env name=previous-commit::$(curl -s https://api.github.com/repos/wix-incubator/rich-content/commits/${{ github.sha }} | jq '.parents[0].sha' | sed 's/\"//g')"
      # - name: Get previous commit status
      #   id: previous-status
      #   run: |
      #     echo "::set-output name=previous-status::$(echo $(curl -s https://api.github.com/repos/wix-incubator/rich-content/commits/${{ steps.previous-commit.outputs.previous-commit }}/check-suites\?app_id\=15368 -H "Accept: application/vnd.github.antiope-preview+json" | jq '.check_suites | [.[] | select( .head_branch == "master") | .conclusion] | .[:2] | .[]' | sed 's/\"//g'))"
      - name: Get previous commit status
        id: previous-status
        run: |
          echo "::set-env name=previous-status::$(echo $(curl -s https://api.github.com/repos/wix-incubator/rich-content/actions/workflows/tests.yml/runs\?branch\=master | jq '.workflow_runs | .[] | select( .head_sha == "${{ env.previous-commit }}") | .conclusion' | sed 's/\"//g'))"
      - name: Notify recovery in Slack
        uses: voxmedia/github-action-slack-notify-build@v1.1.1
        if: github.ref == 'refs/heads/master' && env.previous-status == 'failure' && (needs.e2e-visual.result == 'success' && needs.content-state-tests.result == 'success' && needs.lint-and-unit-tests.result == 'success')
        with:
          channel_id: C0124DH7HHS #rich-content-ci
          status: 'RECOVERED  :celebrate:'
          color: '#28a745'
    env:
      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
      Barackos: <@UM3M904MU>
      barboaron: <@UMJ56Q30W>
      danarish4: <@UEBDPSU2W>
      lxgreen: <@U6FQCJ5GT>
      maimonav: <@UMUB0DSLD>
      mharsat: <@UJAAZU58B>
      noamtwix: <@UEBDQJBMG>
      oded-soffrin: <@U0MN7A3H8>
      ShamaySapir: <@UM48FG657>
      shaulgo: <@U0SDJ5S30>
      talevy17: <@UM24QHHAR>
      michael-wix: ' - Michael just added translations'
      wix-babel-ng: " - Michael's bot added new translations :pokesquirtlewalk:"
