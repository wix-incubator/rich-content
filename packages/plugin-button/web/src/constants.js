/* eslint-disable camelcase */
export const BUTTON_TYPE = 'wix-draft-plugin-button';
export const ALIGN_CENTER = 'center';
export const settingsTabValue = 'settings';
export const designTabValue = 'design';
export const BUTTON_TYPES = ['primary', 'secondary'];
export const DEFAULT_COLOR = '#000000';
export const DEFAULT_SELECTION_COLOR = '#000000';
export const COLOR_PICKER_TYPE = Object.freeze({
  TEXT_COLOR: 'textColor',
  BORDER_COLOR: 'borderColor',
  BACKGROUND_COLOR: 'backgroundColor',
});

export const COLORS = Object.freeze({
  color1: '#FFFFFF',
  color2: '#D5D4D4',
  color3: '#000000',
  color4: '#000000',
  color5: '#000000',
  color6: '#ABCAFF',
  color7: '#81B0FF',
  color8: '#0261FF',
  color9: '#0141AA',
  color10: '#012055',
});

export const DEFAULT_CONFIG = {
  alignment: ALIGN_CENTER,
  size: 'small',
  width: 'fit-content',
};

export const buttonPreviews = colors => [
  {
    className: 'button_primary',
    border: '0px solid #' + colors.color8,
    borderRadius: '0px',
    borderWidth: '0px',
    background: colors.color8,
    color: colors.color1,
    borderColor: colors.color8,
  },
  {
    className: 'button_secondary',
    border: '1px solid ' + colors.color8,
    borderRadius: '0px',
    borderWidth: '1px',
    background: colors.color1,
    color: colors.color8,
    borderColor: colors.color8,
  },
  {
    className: 'button_secondary',
    border: '1px solid ' + colors.color8,
    borderRadius: '0px',
    borderWidth: '1px',
    background: colors.color7,
    color: colors.color8,
    borderColor: colors.color8,
  },
  {
    className: 'button_secondary',
    border: '1px solid ' + colors.color8,
    borderRadius: '10px',
    borderWidth: '1px',
    background: colors.color7,
    color: colors.color8,
    borderColor: colors.color8,
  },
  {
    className: 'button_secondary',
    border: '1px solid ' + colors.color8,
    borderWidth: '5px',
    borderRadius: '0px',
    background: colors.color1,
    color: colors.color8,
    borderColor: colors.color8,
  },
];
