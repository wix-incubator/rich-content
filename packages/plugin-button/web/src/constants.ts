/* eslint-disable camelcase */
export const ALIGN_CENTER = 'center';
export const settingsTabValue = 'settings';
export const designTabValue = 'design';
export const BUTTON_TYPES = ['primary', 'secondary'];
export const DEFAULT_COLOR = '#000000';
export const DEFAULT_SELECTION_COLOR = '#000000';
export const COLOR_PICKER_TYPE = Object.freeze({
  TEXT_COLOR: 'textColor',
  BORDER_COLOR: 'borderColor',
  BACKGROUND_COLOR: 'backgroundColor',
});

export const DEFAULT_CONFIG = {
  alignment: ALIGN_CENTER,
  size: 'small',
  width: 'fit-content',
};
