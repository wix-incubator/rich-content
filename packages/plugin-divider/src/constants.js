export const DIVIDER_TYPE = 'wix-draft-plugin-divider';

export const LINE_SINGLE = 'single';
export const LINE_DOUBLE = 'double';
export const LINE_DASHED = 'dashed';
export const LINE_DOTTED = 'dotted';

export const LINE_TYPES = [LINE_SINGLE, LINE_DOUBLE, LINE_DASHED, LINE_DOTTED];

export const ALIGN_LEFT = 'left';
export const ALIGN_CENTER = 'center';
export const ALIGN_RIGHT = 'right';

export const SIZE_SMALL = 'small';
export const SIZE_MEDIUM = 'medium';
export const SIZE_LARGE = 'large';

export const DEFAULTS = {
  type: LINE_SINGLE,
  config: {
    dividerSize: SIZE_LARGE,
    dividerAlignment: ALIGN_CENTER
  }
};
