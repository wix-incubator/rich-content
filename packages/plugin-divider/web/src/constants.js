export const DIVIDER_TYPE = 'wix-draft-plugin-divider';

export const LINE_SINGLE = 'single';
export const LINE_DOUBLE = 'double';
export const LINE_DASHED = 'dashed';
export const LINE_DOTTED = 'dotted';

export const LINE_TYPES = Object.freeze([LINE_SINGLE, LINE_DOUBLE, LINE_DASHED, LINE_DOTTED]);

export const ALIGN_LEFT = 'left';
export const ALIGN_CENTER = 'center';
export const ALIGN_RIGHT = 'right';

export const SIZE_SMALL = 'small';
export const SIZE_MEDIUM = 'medium';
export const SIZE_LARGE = 'large';

export const SIZES = Object.freeze([SIZE_LARGE, SIZE_MEDIUM, SIZE_SMALL]);
export const ALIGNMENTS = [ALIGN_LEFT, ALIGN_CENTER, ALIGN_RIGHT];
export const NO_WRAP = 'nowrap';

export const DEFAULTS = Object.freeze({
  type: LINE_SINGLE,
  config: {
    size: SIZE_LARGE,
    alignment: ALIGN_CENTER,
    textWrap: NO_WRAP,
  },
});
