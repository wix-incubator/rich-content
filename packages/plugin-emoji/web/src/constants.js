import React from 'react';
import FaBell from './icons/FaBell.svg';
import FaFlag from './icons/FaFlag.svg';
import FaHeart from './icons/FaHeart.svg';
import FaPaw from './icons/FaPaw.svg';
import FaSmile from './icons/FaSmile.svg';
import FaPlane from './icons/FaPlane.svg';
import FaUtensils from './icons/FaUtensils.svg';
import FaRegFutbol from './icons/FaRegFutbol.svg';
export const EMOJI_TYPE = 'wix-draft-plugin-emoji';

export const getEmojiGroups = t => [
  {
    title: t('EmojiPlugin_EmojiGroups_People'),
    icon: <FaSmile />,
    category: 'people',
  },
  {
    title: t('EmojiPlugin_EmojiGroups_Nature'),
    icon: <FaPaw />,
    category: 'nature',
  },
  {
    title: t('EmojiPlugin_EmojiGroups_Food'),
    icon: <FaUtensils />,
    category: 'food',
  },
  {
    title: t('EmojiPlugin_EmojiGroups_Activity'),
    icon: <FaRegFutbol />,
    category: 'activity',
  },
  {
    title: t('EmojiPlugin_EmojiGroups_Travel'),
    icon: <FaPlane />,
    category: 'travel',
  },
  {
    title: t('EmojiPlugin_EmojiGroups_Objects'),
    icon: <FaBell />,
    category: 'objects',
  },
  {
    title: t('EmojiPlugin_EmojiGroups_Symbols'),
    icon: <FaHeart />,
    category: 'symbols',
  },
  {
    title: t('EmojiPlugin_EmojiGroups_Flags'),
    icon: <FaFlag />,
    category: 'flags',
  },
];

export const DesktopFlyOutModalStyles = Object.freeze({
  overlay: {
    backgroundColor: 'rgba(0, 0, 0, 0)',
    zIndex: 5,
  },
  content: {
    width: '320px',
    boxSizing: 'border-box',
    height: '364px',
    overflow: 'visible',
    border: 'solid 1px rgba(51, 51, 51, 0.1)',
    paddingRight: '0px',
    paddingLeft: '0px',
    display: 'block',
    borderRadius: '2px',
    position: 'absolute',
    zIndex: 6,
    paddingTop: '20px',
    boxShadow: '0 0 10px 0 rgba(0, 0, 0, 0.06)',
  },
});

export const DEFAULT_CONFIG = {};
