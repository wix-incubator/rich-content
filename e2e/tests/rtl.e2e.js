/*global cy*/
import { DEFAULT_DESKTOP_BROWSERS, DEFAULT_MOBILE_BROWSERS } from '../tests/constants';

describe('rtl', () => {
  beforeEach(() => cy.switchToHebrew());

  afterEach(() => cy.matchContentSnapshot());

  context('desktop', () => {
    before(function() {
      cy.eyesOpen({
        appName: 'RTL',
        testName: this.test.parent.title,
        browser: DEFAULT_DESKTOP_BROWSERS,
      });
    });

    beforeEach(() => cy.switchToDesktop());

    after(() => cy.eyesClose());

    it('render plugin toolbar in rtl', function() {
      cy.loadEditorAndViewer()
        .focusEditor()
        .openSideToolbar();
      cy.eyesCheckWindow(this.test.title);
    });

    it('render text toolbar in rtl', function() {
      cy.loadEditorAndViewer('plain')
        .setSelection(0, 8)
        .get('[data-hook=inlineToolbar]')
        .should('be.visible')
        .get('[data-hook=addPluginFloatingToolbar]')
        .should('be.visible');
      cy.eyesCheckWindow(this.test.title);
    });

    it('render rtl and ltr text correctly', function() {
      cy.loadEditorAndViewer('hebrew');
      cy.eyesCheckWindow(this.test.title);
    });

    it('render external modal in rtl', function() {
      cy.loadEditorAndViewer('images')
        .openImageSettings()
        .get('[data-hook="imageSettingsCaptionInput"]')
        .blur();
      cy.eyesCheckWindow(this.test.title);
    });
  });

  context('mobile', () => {
    before(function() {
      cy.eyesOpen({
        appName: 'RTL',
        testName: this.test.parent.title,
        browser: DEFAULT_MOBILE_BROWSERS,
      });
    });

    beforeEach(() => cy.switchToMobile());

    after(() => cy.eyesClose());

    it('render add plugin modal in rtl', function() {
      cy.loadEditorAndViewer()
        .focusEditor()
        .openAddPluginModal();
      cy.eyesCheckWindow(this.test.title);
    });

    it('render rtl and ltr text correctly', function() {
      cy.loadEditorAndViewer('hebrew');
      cy.eyesCheckWindow(this.test.title);
    });

    it('render external modal in rtl', function() {
      cy.loadEditorAndViewer('images')
        .openImageSettings()
        .get('[aria-label="Cancel"]')
        .blur();
      cy.eyesCheckWindow({ tag: this.test.title, target: 'window', fully: false });
    });
  });
});
