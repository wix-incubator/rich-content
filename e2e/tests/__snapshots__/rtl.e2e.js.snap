exports[`rtl > desktop > render plugin toolbar in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`rtl > desktop > render text toolbar in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "Bring to the table win-win survival strategies to ensure proactive domination.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 3,
      "text": "At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "User generated content in real-time will have multiple touchpoints for offshoring.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 5,
      "text": "Capitalize on low hanging fruit to identify a ballpark value added activity to beta test. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "Override the digital divide with additional clickthroughs from DevOps.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 7,
      "text": "Nanotechnology immersion along the information highway will close the loop on focusing solely on the bottom line.",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`rtl > desktop > render rtl and ltr text correctly #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "לורם איפסום דולור סיט אמט, קונסקטורר אדיפיסינג אלית נולום ארווס סאפיאן - פוסיליס קוויס, אקווזמן צש בליא, מנסוטו צמלח לביקו ננבי, צמוקו בלוקריה שיצמה ברורק. ושבעגט ליבם סולגק. בראיט ולחת צורק מונחף, בגורמי מגמש. תרבנך וסתעד לכנו סתשם השמה - לתכי מורגם בורק? לתיג ישבעס.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`rtl > desktop > render external modal in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 2,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 3,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 5,
      "text": "Bring to the table win-win survival strategies to ensure proactive domination.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 1,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 7,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 8,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 9,
      "text": "At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution.",
      "type": "unstyled"
    }
  ],
  "entityMap": {
    "0": {
      "data": {
        "config": {
          "alignment": "center",
          "showDescription": true,
          "showTitle": true,
          "size": "small"
        },
        "src": {
          "file_name": "8bb438_131a7e1872bc45ec827bb61e56b840fe.jpg",
          "height": 3354,
          "id": "036c6bf6cef5e4409848eb4eb6f80de1",
          "original_file_name": "8bb438_131a7e1872bc45ec827bb61e56b840fe.jpg",
          "width": 2898
        }
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    },
    "1": {
      "data": {
        "config": {
          "alignment": "center",
          "showDescription": true,
          "showTitle": true,
          "size": "content"
        },
        "src": {
          "file_name": "8bb438_83e03311e53a47ce9ba8bda910fdad88.jpg",
          "height": 3526,
          "id": "b12907da5fbb55a31a760e49fb0bb4e6",
          "original_file_name": "8bb438_83e03311e53a47ce9ba8bda910fdad88.jpg",
          "width": 5177
        }
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    }
  }
};

exports[`rtl > mobile > render add plugin modal in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`rtl > mobile > render rtl and ltr text correctly #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "לורם איפסום דולור סיט אמט, קונסקטורר אדיפיסינג אלית נולום ארווס סאפיאן - פוסיליס קוויס, אקווזמן צש בליא, מנסוטו צמלח לביקו ננבי, צמוקו בלוקריה שיצמה ברורק. ושבעגט ליבם סולגק. בראיט ולחת צורק מונחף, בגורמי מגמש. תרבנך וסתעד לכנו סתשם השמה - לתכי מורגם בורק? לתיג ישבעס.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`rtl > mobile > render external modal in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 2,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 3,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 5,
      "text": "Bring to the table win-win survival strategies to ensure proactive domination.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 1,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 7,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 8,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 9,
      "text": "At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution.",
      "type": "unstyled"
    }
  ],
  "entityMap": {
    "0": {
      "data": {
        "config": {
          "alignment": "center",
          "showDescription": true,
          "showTitle": true,
          "size": "small"
        },
        "src": {
          "file_name": "8bb438_131a7e1872bc45ec827bb61e56b840fe.jpg",
          "height": 3354,
          "id": "036c6bf6cef5e4409848eb4eb6f80de1",
          "original_file_name": "8bb438_131a7e1872bc45ec827bb61e56b840fe.jpg",
          "width": 2898
        }
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    },
    "1": {
      "data": {
        "config": {
          "alignment": "center",
          "showDescription": true,
          "showTitle": true,
          "size": "content"
        },
        "src": {
          "file_name": "8bb438_83e03311e53a47ce9ba8bda910fdad88.jpg",
          "height": 3526,
          "id": "b12907da5fbb55a31a760e49fb0bb4e6",
          "original_file_name": "8bb438_83e03311e53a47ce9ba8bda910fdad88.jpg",
          "width": 5177
        }
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    }
  }
};
